packages/builder/src/components/automation/SetupPanel/AutomationBlockSetup.svelte

```diff
@@ -180,54 +253,45 @@
           },
         }
       }
-      const outputs = Object.entries(schema)
-      let bindingIcon = ""
-      let bindingRank = 0
-      if (idx === 0) {
-        bindingIcon = automation.trigger.icon
-      } else if (isLoopBlock) {
-        bindingIcon = "Reuse"
-        bindingRank = idx + 1
-      } else {
-        bindingIcon = allSteps[idx].icon
-        bindingRank = idx - loopBlockCount
+
+      if (idx === 0 && automation.trigger?.event === "app:trigger") {
+        schema = Object.fromEntries(
+          Object.keys(automation.trigger.inputs.fields).map(key => [
```

Comment:
One minor change here would be to change this to `automation.trigger.inputs.fields || []`. I ran this through old automations and it seems that `fields` wasn't being defaulted to an empty array on create and the builder crashes. New automations will add in the empty array.

Reply:
Sorted! Great catch, thanks Dean!